WordPress/src/main/java/org/wordpress/android/ui/main/SitePickerActivity.java

```diff
@@ -508,22 +521,24 @@ private void saveSitesVisibility(Set<SiteRecord> changeSet) {
         List<SiteModel> siteList = new ArrayList<>();
         for (SiteRecord siteRecord : changeSet) {
             SiteModel siteModel = mSiteStore.getSiteByLocalId(siteRecord.getLocalId());
-            if (hiddenSites.contains(siteRecord)) {
-                if (siteRecord.getLocalId() == mCurrentLocalId) {
-                    skippedCurrentSite = true;
-                    currentSiteName = siteRecord.getBlogNameOrHomeURL();
-                    continue;
+            if (siteModel != null) {
```

Comment:
🔍  Nitpick: to avoid nesting for this `null` check, do you think it could make sense to "return" early (actually skip in this case)?

```suggestion
            if (siteModel == null) continue;
```